### ISSUE TITLE
Manage better Pardiso dependency 

### ISSUE THREAD

**Issue by jd-lara** [NREL-Sienna/PowerNetworkMatrices.jl#74]
Consider the following issue from the MKL library 

https://github.com/JuliaLinearAlgebra/MKL.jl/issues/146

https://github.com/JuliaLinearAlgebra/MKL.jl/issues/129

Check the use of https://github.com/JuliaPackaging/Preferences.jl to avoid issues. 

Refs for Mac Users 

https://developer.apple.com/documentation/accelerate/sparse_solvers/sparse_direct_solving_functions_matrix_rhs

**Comment by jd-lara** [comment]
On top of this, some users have been facing challenges with the binaries in windows.

**Comment by gmantegna** [comment]
@jd-lara I am still getting a warning about MKL_jll when I install and load in PowerNetworkMatrices on macOS with Apple Silicon. It seems everything is working fine, but are there plans to address this? Is it indicating something actually wrong or is everything fine? I will note also that MKL itself actually installs fine, when I install that on its own.

<img width="440" alt="Screenshot 2024-07-26 at 2 27 56 PM" src="https://github.com/user-attachments/assets/d65cb666-8a45-4156-a45e-b2583dc3eb1b">

**Comment by ViralBShah** [comment]
macOS aarch64 is not an Intel platform, and hence won't have MKL!

**Comment by gmantegna** [comment]
Yes of course. My question is more specifically: 1) is there something
wrong with the functioning of the code if a user is not on an Intel
platform? (presumably OpenBLAS is used instead or something like that?),
and 2) if the answer to 1) is no, is it really worth throwing a warning? To
me "warning" implies that the code developers think there is potentially
something going wrong with the functioning of the code, if the thing that
triggers the warning is happening.

**Comment by ViralBShah** [comment]
I have been thinking about that too. This is really for the package author to address - because one could use `AppleAccelerate` on apple silicon, and should ideally not load MKL which is a platform specific library.

Perhaps we need a `BLASPreferences.jl` package that helps deal with all this in a more streamlined way.

**Comment by jd-lara** [comment]
@gmantegna we support MKLPardiso for our HPC applications for large scale. Most of the developers from Sienna use Macs and we use the KLU solver since for systems below 15k buses there is no real difference between KLU and MKLPardiso. 

TBH we mostly ignore that warning since we don't use that solver in Mac applications.

**Comment by ViralBShah** [comment]
You could just put an `@static if` platform check around MKL.
